tests/integration/test_async_insert_parallel_parsing/test.py

```diff
@@ -0,0 +1,95 @@
+import logging
+import random
+import timeit
+from itertools import repeat
+from math import floor
+from multiprocessing import Pool
+
+import pytest
+
+from helpers.cluster import ClickHouseCluster
+
+cluster = ClickHouseCluster(__file__)
+
+
+node = cluster.add_instance(
+    "node",
+    main_configs=["configs/zookeeper_config.xml", "configs/config.xml"],
+    user_configs=[
+        "configs/users.xml",
+    ],
+    with_zookeeper=True,
+)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def start_cluster():
+    try:
+        cluster.start()
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+def _generate_values(size, min_int, max_int, array_size_range):
+    gen_tuple = lambda _min_int, _max_int, _array_size_range: (
```

Comment:
Why a lambda here instead of defining a new function (even if it's within _generate_values)